### PR TITLE
feat: add initial implementation for status endpoint

### PR DESCRIPTION
**Pull request by etolbakov**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Implement the endpoint to retrieve runtime and build info.

- retrieving information from env variables (using `build_data`) 

## Checklist

- [X]  I have written the necessary rustdoc comments.
- [X]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link
https://github.com/GreptimeTeam/greptimedb/issues/1650

### CODE REVIEW

**Comment by etolbakov** [comment]
This is the response I receive: 
```sh
❯ curl  http://127.0.0.1:4000/status | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   226  100   226    0     0   9203      0 --:--:-- --:--:-- --:--:-- 11300
{
  "source_time": "2023-06-19T08:03:20Z",
  "commit": "3276667bffb13f202a78630470d23966111a28c6",
  "branch": "feat/status-endpoint",
  "rustc_version": "rustc 1.71.0-nightly (a368898de 2023-05-02)",
  "hostname": "<redacted>",
  "version": "0.4.0"
}
```

**Comment by etolbakov** [comment]
@killme2008 @waynexia 
Could you please have a look when you have free time if the implementation looks ok?

File: src/servers/tests/http/http_handler_test.rs [hunk 1]
```diff
@@ -365,3 +365,18 @@ async fn test_health() {
         expected_json_str
     );
 }
+
+#[tokio::test]
+async fn test_status() {
+    let expected_json = http_handler::StatusResponse {
```
[the remark this answers is not in the document]
Comment (waynexia) [reply src/servers/tests/http/http_handler_test.rs:371]: Looks good 👍

**Review by killme2008 — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**